src/Meshes/csg.ts

```diff
@@ -75,7 +75,7 @@ class Vertex {
     public interpolate(other: Vertex, t: number): Vertex {
         return new Vertex(Vector3.Lerp(this.pos, other.pos, t),
             Vector3.Lerp(this.normal, other.normal, t),
-            Vector2.Lerp(this.uv, other.uv, t),
+            this.uv && other.uv ? Vector2.Lerp(this.uv, other.uv, t) : undefined,

```

Comment:
I know the next line is doing the same, but this way of single-line assignment seems unreadable to me. A simple `if()` would be better IMO

Reply:
Not sure what you mean? May you had something like this in mind: (a && b) && (c=d) (which is indeed not very readable in my opinion)?

Here it is doing:
```javascript
new Vertex(
    Vector3.Lerp(this.pos, other.pos, t),
    Vector3.Lerp(this.normal, other.normal, t),
    this.uv && other.uv ? Vector2.Lerp(this.uv, other.uv, t) : undefined,
    this.vertColor && other.vertColor ? Color4.Lerp(this.vertColor, other.vertColor, t) : undefined
);
```
I think it is readable(?)

The alternate way with `if` would be:
```typescript
public interpolate(other: Vertex, t: number): Vertex {
    let uv = undefined;
    if (this.uv && other.uv) {
        uv = Vector2.Lerp(this.uv, other.uv, t);
    }
    let vertColor = undefined;
    if (this.vertColor && other.vertColor) {
        vertColor = Color4.Lerp(this.vertColor, other.vertColor, t)
    }
    return new Vertex(
        Vector3.Lerp(this.pos, other.pos, t),
        Vector3.Lerp(this.normal, other.normal, t),
        uv,
        vertColor);
}
```

Reply:
It is all good. Personally if feels like too much for a constructor (thou i probably do those myself from time to time). I approved it and will merge it very soon.